redcap/project.py

```diff
@@ -615,6 +615,62 @@ def import_records(
             raise RedcapError(str(response))
         return response
 
+    def import_metadata(
+        self,
+        to_import,
+        format="json",
+        return_format="json",
+        date_format="YMD"
+    ):
+        """
+        Import metadata (DataDict) into the RedCap Project
+
+        Parameters
+        ----------
+        to_import : array of dicts, csv/xml string, ``pandas.DataFrame``
+            :note:
+                If you pass a csv or xml string, you should use the
+                ``format`` parameter appropriately.
+        format : ('json'),  'xml', 'csv'
+            Format of incoming data. By default, to_import will be json-encoded
+        return_format : ('json'), 'csv', 'xml'
+            Response format. By default, response will be json-decoded.
+        date_format : ('YMD'), 'DMY', 'MDY'
+            Describes the formatting of dates. By default, date strings
+            are formatted as 'YYYY-MM-DD' corresponding to 'YMD'. If date
+            strings are formatted as 'MM/DD/YYYY' set this parameter as
+            'MDY' and if formatted as 'DD/MM/YYYY' set as 'DMY'. No
+            other formattings are allowed.
+
+        Returns
+        -------
+        response : dict, str
+            response from REDCap API, json-decoded if ``return_format`` == ``'json'``
+            If successful, the number of imported fields
+        """
+        payload = self.__basepl("metadata")
+        # pylint: disable=comparison-with-callable
+        if hasattr(to_import, "to_csv"):
+            # We'll assume it's a df
+            buf = StringIO()
+            to_import.to_csv(buf, index=False)
+            payload["data"] = buf.getvalue()
+            buf.close()
+            format = "csv"
+        elif format == "json":
+            payload["data"] = json.dumps(to_import, separators=(",", ":"))
+        else:
+            # don't do anything to csv/xml
+            payload["data"] = to_import
```

Comment:
Thanks for being consistent with how the `Project.import_records` imports data. 

Would you be open to adding a private helper method `Project._add_import_data` that basically accomplishes the  highlighted section above? Since the code is _exactly_ the same, it probably makes sense to abstract it at this point.

Reply:
I did add a utility method do initialize the payload for `record` and `metadata`. However since the code was not 100% identical i had to add the data type as another input to that helper function. What do you think about this?